src/Infrastructure.VS/VsInfoService.cs

```diff
@@ -41,18 +40,18 @@ public interface IVsInfoService
     [PartCreationPolicy(CreationPolicy.Shared)]
     internal class VsInfoService : IVsInfoService
     {
+        private readonly Lazy<string> installRootDir;
+
         [ImportingConstructor]
-        public VsInfoService([Import(typeof(SVsServiceProvider))] IServiceProvider serviceProvider)
+        public VsInfoService(IVsUIServiceOperation vSServiceOperation)
```

Comment:
Converted the service to use the new service operation rather than using `IServiceProvider` directly. This class previously didn't handle threading at all; using the new service operation helper fixes that.